### PR TITLE
Fix Fedex multiship

### PR DESCRIPTION
**Pull request by jacobshilitz**
Fedex with multi packages was broken

Fedex Taxes did not show

### CODE REVIEW

File: modules/connectors/fedex/karrio/schemas/fedex/rating_response.py [hunk 1]
```diff
@@ -132,6 +137,7 @@ class ShipmentRateDetailType:
     totalSurcharges: Optional[float] = None
     totalFreightDiscount: Optional[float] = None
     surCharges: List[SurchargeType] = JList[SurchargeType]
+    taxes: List[TaxesType] = JList[TaxesType]
```
Comment (danh91) [inline modules/connectors/fedex/karrio/schemas/fedex/rating_response.py:140]: I couldn't find any references to such a field in the docs or in the responses returned by API.
https://developer.fedex.com/api/en-us/catalog/rate/v1/docs.html

Comment (jacobshilitz) [reply modules/connectors/fedex/karrio/schemas/fedex/rating_response.py:140]: I noticed that the docs are not aligned with the actual response... in fact the totalTaxes is empty

<details>
  <summary>Actual Rate API response (dev mode)</summary>

```json
{
  "transactionId": "febbfa9d-6ca8-4963-9dbc-de890574bf6e",
  "output": {
    "alerts": [
      {
        "code": "ORIGIN.STATEORPROVINCECODE.CHANGED",
        "message": "The origin state/province code has been changed.",
        "alertType": "NOTE"
      },
      {
        "code": "SHIPMENT.CHARGES.NOTE",
        "message": "Shipment level charges have been added to first package.",
        "alertType": "NOTE",
        "parameterList": []
      }
    ],
    "rateReplyDetails": [
      {
        "serviceType": "FEDEX_EXPRESS_SAVER",
        "serviceName": "FedEx Economy",
        "packagingType": "YOUR_PACKAGING",
        "customerMessages": [
          {
            "code": "DIM.APPLIED.NOTE",
            "message": "Dimensional Weight of your package(s) was used to calculate your rate."
          }
        ],
        "ratedShipmentDetails": [
          {
            "rateType": "ACCOUNT",
            "ratedWeightMethod": "DIM",
            "totalDiscounts": 0.0,
            "totalBaseCharge": 37.15,
            "totalNetCharge": 48.17,
            "totalNetFedExCharge": 42.63,
            "shipmentRateDetail": {
              "rateZone": "4",
              "dimDivisor": 139,
              "fuelSurchargePercent": 14.75,
              "totalSurcharges": 5.48,
              "totalFreightDiscount": 0.0,
              "surCharges": [
                {
                  "type": "FUEL",
                  "description": "Fuel Surcharge",
                  "level": "SHIPMENT",
                  "amount": 5.48
                }
              ],
              "taxes": [
                {
                  "type": "HST",
                  "description": "Canada Ontario harmonized sales",
                  "amount": 5.54
                }
              ],
              "pricingCode": "",
              "totalBillingWeight": {
                "units": "KG",
                "value": 2.5
              },
              "dimDivisorType": "COUNTRY",
              "totalDimWeight": {
                "units": "KG",
                "value": 2.5
              },
              "currency": "USD",
              "rateScale": "CA0001F4_20_YOUR_PACKAGING",
              "totalRateScaleWeight": {
                "units": "LB",
                "value": 6.0
              }
            },
            "ratedPackages": [
              {
                "groupNumber": 0,
                "effectiveNetDiscount": 0.0,
                "packageRateDetail": {
                  "rateType": "PAYOR_ACCOUNT_SHIPMENT",
                  "ratedWeightMethod": "DIM",
                  "baseCharge": 37.15,
                  "netFreight": 37.15,
                  "totalSurcharges": 5.48,
                  "netFedExCharge": 42.63,
                  "totalTaxes": 5.54,
                  "netCharge": 48.17,
                  "totalRebates": 0.0,
                  "billingWeight": {
                    "units": "KG",
                    "value": 2.5
                  },
                  "totalFreightDiscounts": 0.0,
                  "surcharges": [
                    {
                      "type": "FUEL",
                      "description": "Fuel Surcharge",
                      "level": "SHIPMENT",
                      "amount": 5.48
                    }
                  ],
                  "dimWeight": {
                    "units": "KG",
                    "value": 2.5
                  },
                  "currency": "USD"
                },
                "sequenceNumber": 1
              }
            ],
            "currency": "USD"
          },
          {
            "rateType": "PREFERRED_CURRENCY",
            "ratedWeightMethod": "DIM",
            "totalDiscounts": 0.0,
            "totalBaseCharge": 50.87,
            "totalNetCharge": 65.95,
            "totalNetFedExCharge": 58.37,
            "shipmentRateDetail": {
              "rateZone": "4",
              "dimDivisor": 139,
              "fuelSurchargePercent": 14.75,
              "totalSurcharges": 7.5,
              "totalFreightDiscount": 0.0,
              "surCharges": [
                {
                  "type": "FUEL",
                  "description": "Fuel Surcharge",
                  "amount": 7.5
                }
              ],
              "taxes": [
                {
                  "type": "HST",
                  "description": "Canada Ontario harmonized sales",
                  "amount": 7.58
                }
              ],
              "pricingCode": "",
              "totalBillingWeight": {
                "units": "KG",
                "value": 2.5
              },
              "dimDivisorType": "COUNTRY",
              "totalDimWeight": {
                "units": "KG",
                "value": 2.5
              },
              "currency": "CAD",
              "rateScale": "CA0001F4_20_YOUR_PACKAGING",
              "totalRateScaleWeight": {
                "units": "LB",
                "value": 6.0
              }
            },
            "ratedPackages": [
              {
                "groupNumber": 0,
                "effectiveNetDiscount": 0.0,
                "packageRateDetail": {
                  "rateType": "PREFERRED_ACCOUNT_SHIPMENT",
                  "ratedWeightMethod": "DIM",
                  "baseCharge": 50.87,
                  "netFreight": 50.87,
                  "totalSurcharges": 7.5,
                  "netFedExCharge": 58.37,
                  "totalTaxes": 7.58,
                  "netCharge": 65.95,
                  "totalRebates": 0.0,
                  "billingWeight": {
                    "units": "KG",
                    "value": 2.5
                  },
                  "totalFreightDiscounts": 0.0,
                  "surcharges": [
                    {
                      "type": "FUEL",
                      "description": "Fuel Surcharge",
                      "amount": 7.5
                    }
                  ],
                  "dimWeight": {
                    "units": "KG",
                    "value": 2.5
                  },
                  "currency": "CAD"
                },
                "sequenceNumber": 1
              }
            ],
            "currency": "CAD"
          }
        ],
        "operationalDetail": {
          "ineligibleForMoneyBackGuarantee": false,
          "astraDescription": "XS",
          "airportId": "YYZ",
          "serviceCode": "20"
        },
        "signatureOptionType": "SERVICE_DEFAULT",
        "serviceDescription": {
          "serviceId": "EP1000000013",
          "serviceType": "FEDEX_EXPRESS_SAVER",
          "code": "20",
          "names": [
            {
              "type": "long",
              "encoding": "utf-8",
              "value": "FedEx Economy"
            },
            {
              "type": "long",
              "encoding": "ascii",
              "value": "FedEx Economy"
            },
            {
              "type": "medium",
              "encoding": "utf-8",
              "value": "FedEx Economy"
            },
            {
              "type": "medium",
              "encoding": "ascii",
              "value": "FedEx Economy"
            },
            {
              "type": "short",
              "encoding": "utf-8",
              "value": "XS"
            },
            {
              "type": "short",
              "encoding": "ascii",
              "value": "XS"
            },
            {
              "type": "abbrv",
              "encoding": "ascii",
              "value": "XS"
            }
          ],
          "serviceCategory": "parcel",
          "description": "FedEx Economy",
          "astraDescription": "XS"
        }
      }
    ],
    "quoteDate": "2024-07-18",
    "encoded": false
  }
}
```


  
</details>

Comment (danh91) [reply modules/connectors/fedex/karrio/schemas/fedex/rating_response.py:140]: On that note, can you tell me how you did that API call. Was it in sandbox or FedEx production API?

I saw that you disabled the mock data but when I tried using the sandbox API again it always returned errors when you send something different from the mock data?

What am I missing?🤔

Comment (jacobshilitz) [reply modules/connectors/fedex/karrio/schemas/fedex/rating_response.py:140]: It use the sandbox API, and I also tested with production to make sure its the same

By me the sandbox API allows any data I put in just like the production. perhaps your sandbox API key is restricted? 🤔

**Pull request merged**